og.module

```diff
@@ -117,6 +120,26 @@ function og_entity_insert(EntityInterface $entity) {
 }
 
 /**
+ * Implements hook_entity_field_access().
+ *
+ * Hide group-audience fields from user's edit profile for non-privileged users.
+ */
+function og_entity_field_access($operation, FieldDefinitionInterface $field_definition, AccountInterface $account, FieldItemListInterface $items = NULL) {
+  // User is anonymous, and user register might try to add the group-audience
+  // field.
+  if (($field_definition->getTargetEntityTypeId() !== 'user') || ($operation !== 'edit') || !Og::isGroupAudienceField($field_definition) || $account->isAnonymous()) {
```

Comment:
I prefer to multiple returns like in D7 - it's easier on my brain to understand it :)